notebooks/214-grammar-correction/214-grammar-correction.ipynb

```diff
@@ -0,0 +1,500 @@
+     ]
+    }
+   ],
+   "source": [
+    "from pathlib import Path\n",
+    "from transformers import pipeline, AutoTokenizer\n",
+    "from optimum.intel.openvino import OVModelForSeq2SeqLM, OVModelForSequenceClassification"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6131a0ec-654e-435e-a668-55ad33cff74b",
+   "metadata": {},
+   "source": [
+    "### Grammar Checker"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "47af0ecf-99ff-4852-bfaa-6692caeaca21",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "grammar_checker_model_id = \"textattack/roberta-base-CoLA\"\n",
+    "grammar_checker_dir = Path(\"roberta-base-cola\")\n",
+    "grammar_checker_tokenizer = AutoTokenizer.from_pretrained(grammar_checker_model_id)\n",
+    "\n",
+    "if grammar_checker_dir.exists():\n",
+    "    grammar_checker_model = OVModelForSequenceClassification.from_pretrained(grammar_checker_dir)\n",
+    "else:\n",
+    "    grammar_checker_model = OVModelForSequenceClassification.from_pretrained(grammar_checker_model_id, from_transformers=True)\n",
+    "    grammar_checker_model.save_pretrained(grammar_checker_dir)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "482a5d75-916a-4363-bf24-5b642a6bf437",
+   "metadata": {},
+   "source": [
+    "Let's check model work using inference pipeline for `text-classification` task. You can find more information about usage Hugging Face inference pipelines in this [tutorial](https://huggingface.co/docs/transformers/pipeline_tutorial)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "90e48d59-9eea-4962-ac9a-fc9a6330b406",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "input text: They are moved by salar energy\n",
+      "predicted label: contains_errors, score: 0.88\n"
+     ]
+    }
+   ],
+   "source": [
+    "input_text = \"They are moved by salar energy\"\n",
+    "grammar_checker_pipe = pipeline(\"text-classification\", model=grammar_checker_model, tokenizer=grammar_checker_tokenizer)\n",
+    "result = grammar_checker_pipe(input_text)[0]\n",
+    "print(f\"input text: {input_text}\")\n",
+    "print(f'predicted label: {\"contains_errors\" if result[\"label\"] == \"LABEL_1\" else \"no errors\"}, score: {result[\"score\"] :.2}')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5c4e358c-bbf8-4ea8-9b19-d8616c41562d",
+   "metadata": {},
+   "source": [
+    "Great! Looks like, the model can detect errors in our sample."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cdba3c17-9f94-4d1c-afae-39c857caf5af",
+   "metadata": {
+   },
+   "source": [
+    "### Grammar Corrector\n",
+    "\n",
+    "The steps for loading the Grammar Corrector model will be very similar, except used model class. As FLAN-T5 is a sequence-to-sequence text generation model, we should use `OVModelForSeq2SeqLM` class and `text2text-generation` pipeline for running it."
```

Comment:
```suggestion
    "The steps for loading the Grammar Corrector model are very similar, except for the model class that is used. Because FLAN-T5 is a sequence-to-sequence text generation model, we should use the `OVModelForSeq2SeqLM` class and the `text2text-generation` pipeline to run it."
```